docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+
+Below is the partial class diagram detailing the design of the Add Item Command Class as well as its interactions with the various other classes required to execute the Add Item Command.
+![alt text](team/SiewYangZhi_addItemCommand/AddItemCommandClassDiagram.png)
+Step 1: When the user enters the command `Add Item Toilet Roll / 15`, the `Duke` class would pass the user input to the `Command Parser` Class.
+
+Step 2: The `Command Parser` Class would run the `parse` method and create an `AddItemCommand` object containing an `Item` object which is made up of the name of the item to add as well as its pax which are found within the user input.
+
+Step 3: The `AddItemCommand` class would be passed back to the `Duke` class.
+
+Step 4: The `execute` method of the `AddItemCommand` class would be run and the `ItemList` Object would be extracted from the `List Container` object and the `AddItemCommand` object would pass its Item object to the `ItemList` Object's `addItemToList` method.
+
+Step 5: The `addItemToList` method would then call the `checkForItemDuplicates` method to check if there are any items within the item list with the same name as the item we are about to add.
+
+Step 6:I f there are is an item with a matching name found, we would then call the `UI` object and execute the `UI`'s `printItemAlreadyInTheListErrorMessage` method which would print a message informing the user that the item the user wants to add is already found within the item list and hence nothing would be added. Step 7 and 8 are skipped.
+
+Step 7: If there are no items with a matching name found, we would add the `Item` object to its ArrayList of `Item` objects called listOfItems. The `AddItemCommand` then call the `UI` and execute the `UI`'s `printAddItemAcknowledgementMessage` method which would print an acknowledgement message to the user informing him that the item has been added into the item list.
+
+Step 8: The `ItemListFileManager` object would be called and we would pass the item list to its `writeItemListToFile` method to write the updated item list into the file ListFolder/ItemList.txt.
+
+## Implementation
+
+#### Assign a housekeeper to a room
+The assignment of housekeeper to a room is facilitated by `AssignHousekeeperCommand`. This command objects uses `AssignmentMap`, which keeps a `HashMap`, to help us map a `Room` object to a `Housekeeper` object. 
+
+``
+Map<Room, Housekeeper> map = new HashMap<>();
+``
+
+The following steps exhibit how `AssignHousekeeperCommand` 
+and `AssignmentMap` fit together in the structure and how the mechanism for adding an assinment behaves at each step.
+
+__Step 1.__ The user launches the application. In the `Duke` class, an empty instance of the `AssignmentMap` class,
+called `assignmentMap`, is created.
+
+![Step 1](team/aiman_assignment/step1.png)
```

Comment:
Is it necessary to italicize the `Duke` label here?

Also, if you're not adding any attributes to the classes, would it be fine to remove the boxes that would contain them? (So, only a box containing class names would be left.)
![image](https://user-images.githubusercontent.com/8340533/160964745-690f598b-a978-4f96-9033-e1f9ae4c79f4.png)